src/infrastructure/services/api-clients.ts

```diff
@@ -50,10 +50,14 @@ export class GaiaServiceApiClient {
         };
 
         const cacheTTL = filename === UPLOADABLE_JSON_FILES.CLIENT_CONFIG ? 3600 : undefined;
-        const responseBody: any = await cachedFunctionCall(cacheStorage, options.url, cacheTTL, httpJSONRequest, [options], async (data) => {
-            return Boolean(filename !== UPLOADABLE_JSON_FILES.CLIENT_CONFIG || data.indexOf("BlobNotFound") > 0 || data.indexOf("NoSuchKey") > 0);
-        });
-        return responseBody;
+        try {
+            const responseBody: any = await cachedFunctionCall(cacheStorage, options.url, cacheTTL, httpJSONRequest, [options], async (data) => {
+                return Boolean(filename !== UPLOADABLE_JSON_FILES.CLIENT_CONFIG || data.indexOf("BlobNotFound") > 0 || data.indexOf("NoSuchKey") > 0);
+            });
+            return responseBody;
+        } catch (error) {
+            throw new BaseError(error, "Error while retrieving from Gaia");
```

Comment:
There is already a PackageErrorCode for these kind of failures which should be used